### PR TITLE
Structuring env vars

### PR DESCRIPTION
**Pull request by OleanjiKingCode**
# Structuring env vars

The structure can be seen in the .env.sample file

## Linked issues

closes https://github.com/EveripediaNetwork/issues/issues/1248

### CODE REVIEW

**Comment by Emmanueldmlr** [comment]
@OleanjiKingCode , why do we have googleAnalyticsScript twice?
image

File: src/env.mjs [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  REVALIDATE_PAGES_SECRET: z.string(),
+  OPENAI_API_KEY: z.string().optional(),
+  PINATA_KEY: z.string().optional(),
+  PINATA_SECRET: z.string().optional(),
+})
+
+const client = z.object({
+  NEXT_PUBLIC_IS_PRODUCTION: z.string(),
+  NEXT_PUBLIC_INFURA_ID: z.string(),
+  NEXT_PUBLIC_ALCHEMY_API_KEY: z.string(),
```
Comment (Emmanueldmlr) [inline src/env.mjs:14]: I think this should be moved to the server, we also don't want to expose this I guess.

File: src/env.mjs [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  NEXT_PUBLIC_PINATA_GATEWAY_BASE_URL: z.string().url(),
+  NEXT_PUBLIC_MAGIC_LINK_API_KEY: z.string(),
```
Comment (Emmanueldmlr) [inline src/env.mjs:16]: We should move this to the server too.

File: src/env.mjs [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+  NEXT_PUBLIC_ENS_RPC: z.string().url(),
+  NEXT_PUBLIC_GOOGLE_ANALYTICS: z.string(),
```
Comment (Emmanueldmlr) [inline src/env.mjs:18]: move this to the server too.

**Comment by OleanjiKingCode** [comment]
> @OleanjiKingCode , why do we have googleAnalyticsScript twice? image

Not sure about that i just changed the env vars used there, ill check it though

File: src/env.mjs (hunk 3 above)
Comment (kesar) [reply src/env.mjs:18]: i think googel analytics need to be on the client

File: src/env.mjs (hunk 1 above)
Comment (OleanjiKingCode) [reply src/env.mjs:14]: expose??

Comment (kesar) [reply src/env.mjs:14]: not sure if it will work if you move it to backend.. do testing if you do these changes

**Review by Royal-lobster — APPROVED** [review]

**Pull request merged**